### ISSUE TITLE
[Bug]: pessimistic transaction and rc mode: drop table should reported table not exists

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#9965]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):767517b81caf4a05b3c0885eaadb52d30700ce94
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

drop table should reported table not exists,but now execute success
![image](https://github.com/matrixorigin/matrixone/assets/113406637/b8839b42-aae3-4f8f-bfed-1a10ec45e82d)
![image](https://github.com/matrixorigin/matrixone/assets/113406637/2251ee35-fc41-4540-b336-eece17feabd1)

### Steps to Reproduce

```Markdown
drop table if exists atomic_table_11;
create table atomic_table_11(c1 int,c2 varchar(25));
insert into atomic_table_11 values (3,"a"),(4,"b"),(5,"c");
begin;
-- @session:id=2&user=sys:dump&password=111
drop table atomic_table_11;
-- @session
drop table atomic_table_11;
commit;
```

**Comment by heni02** [comment]
@zhangxu19830126 plz confirm this issue

**Comment by sukki37** [comment]
@zhangxu19830126 Pls kindly check and if there is any doubt you can talk with @heni02.

**Comment by zhangxu19830126** [comment]
@volgariver6 it may fixed in you other drop table issue with pessimistic transaction and rc mode

**Comment by volgariver6** [comment]
这两天提新的pr

**Comment by volgariver6** [comment]
pr需要再完善一下

**Comment by volgariver6** [comment]
fixed by #10045

**Comment by heni02** [comment]
confirm,closed
commit id: a65a505f3549160c83057125303e5f8f442a8a1b
![image](https://github.com/matrixorigin/matrixone/assets/113406637/bdb2b2e4-f263-4bab-b7b7-2a3a4ef35560)
